airbyte-integrations/connectors/source-amazon-mws/source_amazon_mws/client.py

```diff
@@ -0,0 +1,171 @@
+import json
+import time
+import pkgutil
+from datetime import datetime, timedelta, date
+from dateutil.relativedelta import *
+from typing import DefaultDict, Dict, Generator
+
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.models import AirbyteStream, AirbyteMessage, AirbyteStateMessage, AirbyteRecordMessage, Type
+
+from .amazon import AmazonClient
+
+
+class BaseClient:
+    GET_FLAT_FILE_ALL_ORDERS_DATA_BY_ORDER_DATE_GENERAL = "GET_FLAT_FILE_ALL_ORDERS_DATA_BY_ORDER_DATE_GENERAL"
+    _ENTITIES = [GET_FLAT_FILE_ALL_ORDERS_DATA_BY_ORDER_DATE_GENERAL]
+    MAX_SLEEP_TIME = 512
+
+    def __init__(self, refresh_token: str, lwa_app_id: str, lwa_client_secret: str, aws_secret_key: str, aws_access_key: str, role_arn: str, start_date: str):
+        self.credentials = dict(
+            refresh_token=refresh_token,
+            lwa_app_id=lwa_app_id,
+            lwa_client_secret=lwa_client_secret,
+            aws_secret_key=aws_secret_key,
+            aws_access_key=aws_access_key,
+            role_arn=role_arn,
+        )
+        self.start_date = start_date
+
+    def _get_amazon_client(self):
+        _amazon_client = AmazonClient(credentials=self.credentials)
+        return _amazon_client
+
+    def check_connection(self):
+        CreatedAfter = (datetime.utcnow() - timedelta(days=1)).isoformat()
+        _amazon_client = self._get_amazon_client()
+        _amazon_client.fetch_orders(CreatedAfter)
+
+    def get_streams(self):
+        streams = []
+        for entity in self._ENTITIES:
+            raw_schema = json.loads(pkgutil.get_data(
+                self.__class__.__module__.split(".")[0], f"schemas/{entity}.json"))
+            streams.append(AirbyteStream.parse_obj(raw_schema))
+        return streams
+
+    def read_reports(self, logger: AirbyteLogger,  entity: str, state: DefaultDict[str, any]) -> Generator[AirbyteMessage, None, None]:
+        cursor_field = "purchase-date"
+        stream_name = entity
+        purchased_date = self._get_cursor_or_none(
```

Comment:
we can simplify a lot of the logic here by doing the following: 

1. remove `_get_default_params` method
2. `purchased_date = self._get_cursor_or_none(state, stream_name, cursor_field) or self.start_date`
3. use `purchased_date` wherever you use `default_cursor_value` -- now `default_cursor_value` is not needed